plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/S3SelectPushdownResultBuilder.java

```diff
@@ -0,0 +1,93 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3select;
+
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.Type;
+import io.trino.testing.MaterializedResult;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+/*
+    Utility class for testing S3 Select pushdown with larger files, that prepares the MaterializedResult based
+    on input files. Only working with BIGINT and VARCHAR fields (VARCHAR is used as a replacement for STRING,
+    as the unbounded VARCHAR behaves the same in the MaterializedRow / MaterializedResult).
+*/
+public class S3SelectPushdownResultBuilder
+{
+    private static final Set<Type> TESTING_TYPES = Set.of(BIGINT, VARCHAR);
+    private String[] files;
+    private Type[] types;
+    private String delimiter;
+
+    public S3SelectPushdownResultBuilder(String[] files, Type[] types, String delimiter)
+    {
+        assertTrue(Arrays.stream(types).allMatch(TESTING_TYPES::contains),
+                "S3SelectPushdownResultBuilder failed to initialize: Supported testing types are {BIGINT, VARCHAR}.");
+        this.files = files;
+        this.types = types;
+        this.delimiter = delimiter;
+    }
+
+    public MaterializedResult buildExpectedPushdownResult(ConnectorSession session) throws IOException
+    {
+        assertNotNull(files, "S3SelectPushdownResultBuilder failed to initialize: No files.");
+        assertNotNull(types, "S3SelectPushdownResultBuilder failed to initialize: No types.");
+
+        MaterializedResult.Builder builder = MaterializedResult.resultBuilder(session, types);
+
+        for (String file : files) {
+            try (BufferedReader br = new BufferedReader(new FileReader(file, UTF_8))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    // Split will remove trailing whitespaces if we do not use the second parameter
+                    String[] values = line.split(delimiter, -1);
+                    Object[] row = mapRow(values, types);
+                    builder.row(row);
+                }
+            }
+        }
+
+        return builder.build();
+    }
+
+    private Object[] mapRow(String[] values, final Type[] types)
+    {
+        return IntStream.range(0, types.length)
```

Comment:
Considering we use BigInt, should this be LongStream instead?

Reply:
I'm not sure I get your comment, the IntStream is basically iterating through the interval [0, numberOfColumns)